code/Forms/UsedOnTable.php

```diff
@@ -64,41 +72,341 @@ public function getSchemaDataDefaults()
     }
 
     /**
+     * An extension hook can be used to do the following:
+     * a) Exclude DataObjects of a particular type of class from showing in results
+     * b) Link ancestors of a DataObject so they show on the same row of results
+     *
+     * Example php:
+     *
+     * namespace My\App\Extensions;
+     *
+     * use SilverStripe\Core\Extension;
+     * use SilverStripe\ORM\DataObject;
+     *
+     * class UsedOnTableExtension extends Extension
+     * {
+     *
+     *     public function updateUsageExcludedClasses(array &$excludedClasses)
+     *     {
+     *         $excludedClasses[] = MyDataObjectToExclude::class;
+     *     }
+     *
+     *     public function updateUsageAncestorDataObjects(array &$ancestorDataObjects, DataObject $dataObject)
+     *     {
+     *         if (!($dataObject instanceof MyDataObjectThatIWantToLink)) {
+                  return;
+               }
+     *         $parentObjectIWantToIgnore = $dataObject->MyParentComponent();
+     *         $grandParentObjectIWantToLink = $parentObjectIWantToIgnore->MyParentComponent();
+     *         // Add $grandParentObjectIWantToLink to ancestors, but not $parentObjectIWantToIgnore
+     *         $ancestorDataObjects[] = $grandParentIWantToLink;
+     *     }
+     * }
+     *
+     * Example yml:
+     *
+     * SilverStripe\Admin\Forms\UsedOnTable:
+     *   extensions:
+     *     - My\App\Extensions\UsedOnTableExtension
+     *
      * @param HTTPRequest $request
      * @return HTTPResponse $response
+     *
+     * @throws \ReflectionException
      */
     public function usage(HTTPRequest $request)
     {
         $record = $this->getRecord();
-        $usage = $record->findOwners();
 
+        // Relationship types has_many and belongs_to are not required
```

Comment:
This method does not seem to be taking into account ownership. So you'll get records that do not own the file. I don't think this was the intention of the card.

Can you explain what this updated method does that `findOwners` didn't do?

Reply:
Have you considered the following scenarios?
- polymorphic relations
- many-to-many-througth
- DataObject inheritance ... what happens if DataObject A has a relation to File and DataObject B extends DataObject A?

Reply:
If we do decide this updated logic is necessary, I think it should on a lower level class.

Reply:
>> Split it up in smaller more manageable private method.
Can do, though maybe do alittle later, need to agree on the fundamentals first.

>> There's already methods on DataObjects for interacting with relations. This seems like it's reinventing the wheel.
There might be.  I did look though I did not find what I needed

>> This method does not seem to be taking into account ownership
The $owns / findOwners() approach doesn't return a complete set of results, it only returns a subset of results and misses a bunch of things like File is attached to a DataObject on a has_one relationship that didn't go through the ownership API.  Didn't seem like much point doing any work on a feature where it was kind of non-functional in its current state. That's why I redid findUsage().

>> many-to-many-through
Yes.  

>> polymorphic relations / DataObject inheritance
I believe so, it's covered by the block `if (!($record instanceof $componentClass)) .. continue` where $componentClass is will usually be File, sometimes as subclass e.g. MyImage.  `instanceof` works with subclasses

>> I think it should on a lower level class.
Any suggestions?  I made it agnostic so it'll work on DataObject not just File

Reply:
When we publish recursively or restore objects, we only interact with relations that are owned. For example, if your page has an unown relation to File and it gets added to a campaign, the related file won't be added to the campaign. It's not clear to me that returning every single reference to the file was the intention here. 

It's probably worth validating this wit PUX first.

Reply:
Would seem weird not to IMO, I read "Used on" to mean "This file is used on these Objects", not a subset of this which is effectively "When I publish these Objects, this File will also get published".  I assumed the intention is the table is to let you know when it's safe to delete a file because it's not used anywhere.

@sachajudd Do you know if the File "Used on table" is supposed to:
a) Show relations only to Objects (Pages + other Objects such as File Blocks) that are "Owned" by the file - means that when you click "publish" the file will also be published
b) Show relations to all Objects that make use of the File anywhere on the site

Reply:
b) sounds sensible to me. Here are the [designs](https://projects.invisionapp.com/dsm/silver-stripe/silverstripe-cms/nav/build/asset/componentContainers/5f152068f6772b33db2e52b0/tab/design?mode=preview) which outline what we were initially thinking.

Reply:
[I've added my two cents on this in the main issue too.](https://github.com/silverstripe/silverstripe-admin/issues/1098#issuecomment-705262381)

Reply:
Following the discussions, seems must people favour displaying everything regardless of ownership.

My points about splitting this into smaller methods and moving the logic to a better location remain. The current logic for finding owners is in versioned, but that's mostly an artefact that ownership is primarily used when publishing DataObject.

Probably would make sense to move the logic on DataObject or DataObjectSchema.